### ISSUE TITLE
Alert Settings?

### ISSUE THREAD

**Issue by jklarfeld** [projectjellyfish/api#574]
On the settings page (iOS), there are a number of switches to control what type of Alerts you get on device.  There is no available outlet I can find in the web service to `GET`/`POST` this information.  One would think it would be in `/staff/:id/settings`, but there is currently no documentation for this page at `/documentation`, and a quick `GET` to this endpoint returns an empty array.

**Comment by calebthompson** [comment]
Are you prepending the API namespace?

```
staff_settings GET      /api/v1/staff/:staff_id/settings(.:format)                staff_settings#index {:format=>:json}
                                           POST     /api/v1/staff/:staff_id/settings(.:format)                staff_settings#create {:format=>:json}
                             staff_setting GET      /api/v1/staff/:staff_id/settings/:id(.:format)            staff_settings#show {:format=>:json}
                                           PATCH    /api/v1/staff/:staff_id/settings/:id(.:format)            staff_settings#update {:format=>:json}
                                           PUT      /api/v1/staff/:staff_id/settings/:id(.:format)            staff_settings#update {:format=>:json}
                                           DELETE   /api/v1/staff/:staff_id/settings/:id(.:format)            staff_settings#destroy {:format=>:json}
```

Routes are defined here for future reference: https://github.com/projectjellyfish/api/blob/master/config/routes.rb#L44

I don't know if you have a local instance of the app running, but if you do you can also run `rake routes | grep staff` to see, for example, all routes related to staff.

**Comment by jklarfeld** [comment]
https://jellyfish-core-build.dpi.bah.com/documentation/1.0/staff_settings/index.html
https://jellyfish-core-build.dpi.bah.com/documentation/1.0/staff_settings/show.html


I assumed that's where this stuff was meant to go, but that still doesn't explain how to use it. For example: If I wanted to receive "Critical" alerts but not "Warning" alerts, what needs to go in the POST? It is normally blank unless someone has set this up?

tl;dr: I know there's a route there, there's nothing explaining how to use it (i.e. Documentation), can you re-open this?

**Comment by calebthompson** [comment]
Yeah sorry, I misread the question. Is this what you're looking for? https://jellyfish-core-build.dpi.bah.com/documentation/1.0/user_setting_options.html

**Comment by calebthompson** [comment]
@jklarfeld wrong thread. Context:

That's closer, but it doesn't really explain how to control Alerts, I mean, I can't send a `POST` with a JSON body that looks like:
```
{ 
     "alerts" : "Hey send less alerts and stuff please"
}
```

The same goes for getting the settings.  I have to assume there are some default alert settings, otherwise you'd never get alerts.


-----

Haha, you totally should. I'll add a special case.

So looking at the first example data:

```
  {
    "id": 5,
    "label": "first label",
    "field_type": "text_area",
    "help_text": "click here",
    "options": "{ \"JSON PAYLOAD\": \"DATA\" }",
    "required": true,
    "created_at": "2015-01-08T13:35:59.737Z",
    "updated_at": "2015-01-08T13:35:59.737Z",
    "deleted_at": null
  }
```

And the POST api for /staff_settings: https://jellyfish-core-build.dpi.bah.com/documentation/1.0/staff_settings.html

We want to POST with the data "name" and "value". We get back a field_type for the options:

```
The field type (text, textarea, radio, checkbox). Varchar(100)
```

So that info tells us a little about values we can expect, and how those input should be formed. text & textareas will just be blobs of text that you can pass, radio will also be text, but preformatted, and check boxes probably have to be an array of strings. The `options` probably defined the keys we expect and display data. I'm not sure what the format is coming out.

So what we probably want for the POST to create a setting is:

```json
{
  "name": "send me emails, yo",
  "value": "false"
}
```

I don't have a convenient way of pulling these out and there's not any seeds. Does the user_setting_options endpoint return anything for you? This is another thing that really doesn't seem to be hooked up in the angular app unfortunately, so it's possible that even if you set any of these values they wouldn't take effect. All I see for them is CRUD, they're not being referenced to decide how many alerts I want Oo.

cc @stackus and @mafernando who mostly wrote those two controllers. Am I missing anything? Are user settings another thing we may just want to remove and reimplement when we are ready to start using them?

**Comment by stackus** [comment]
I'd say yes to remove/ignore until later.

**Comment by mafernando** [comment]
I agree with @stackus - lets ignore user settings for now. We should cycle back with @chriskacerguis to understand how this fits into the road map.
